noir-projects/aztec-nr/aztec/src/test/mocks/mock_note.nr

```diff
@@ -79,7 +79,7 @@ impl NoteInterface<MOCK_NOTE_LENGTH> for MockNote {
 
 impl Eq for MockNote {
     fn eq(self, other: Self) -> bool {
-        (self.header == other.header) & (self.value == other.value)
+        self.value == other.value
     }
 }
```

Comment:
Unrelated but can't we derive this?

Reply:
We could and I did it in https://github.com/AztecProtocol/aztec-packages/pull/11942/commits/845710839aaa2f4288ef0562e6b7b44bef13e2fc

But I was hesitant with doing this because in [my other PR](https://github.com/AztecProtocol/aztec-packages/pull/11774) it caused issue with compilation times. Shouldn't be an issue here though as there is no nesting.